server/services/ebayScraper.js

```diff
@@ -0,0 +1,25 @@
+const puppeteer = require("puppeteer");
+
+const scrapeEbay = async (url) => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  await page.goto(url);
+
+  const [el] = await page.$x('//*[@id="icImg"]');
+  const src = await el.getProperty("src");
+  const imageUrl = await src.jsonValue();
+
+  const [el2] = await page.$x('//*[@id="itemTitle"]/text()');
+  const title = await el2.getProperty("textContent");
+  const titleTxt = await title.jsonValue();
+
+  const [el3] = await page.$x('//*[@id="prcIsum"]');
+  const price = await el3.getProperty("textContent");
+  const priceTxt = await price.jsonValue();
```

Comment:
Interesting... I didn't know you could use XPath with puppeteer. Cool.
This is fine-tuning, but it looks like you have a lot of repeated code here. These three blocks are doing the same thing (given an xpath expression and a property name, they return the property's json value) 😉